include/node.hpp

```diff
@@ -89,9 +88,93 @@ output_ports(Self *self) noexcept {
     (std::make_index_sequence<traits::node::output_ports<Self>::size>());
 }
 
-// Ports can either be a list of ports instances,
-// or two typelists containing port instances -- one for input
-// ports and one for output ports
+/**
+ * @brief The 'node<Derived>' is a base class for blocks that perform specific signal processing operations. It stores
+ * references to its input and output 'ports' that can be zero, one, or many, depending on the use case.
+ * As the base class for all user-defined nodes, it implements common convenience functions and a default public API
+ * through the Curiously-Recurring-Template-Pattern (CRTP). For example:
+ * @code
+ * struct user_defined_block : node<user_defined_block> {
+ *   IN<float> in;
+ *   OUT<float> out;
+ *   // implement one of the possible work or abstracted functions
+ * };
+ * ENABLE_REFLECTION(user_defined_block, in, out);
```

Comment:
The documentation is missing for the other nice macro which would work for the user-defined block with several template parameters like the one below (but with member ports).

Reply:
done... added a commented and fully documented w/ examples where the macros are being declared.